client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -891,9 +893,12 @@ export class PlanFeatures2023Grid extends Component<
 
 const withIsLargeCurrency = ( Component: LocalizedComponent< typeof PlanFeatures2023Grid > ) => {
 	return function ( props: PlanFeatures2023GridType ) {
-		const isLargeCurrency = useIsLargeCurrency(
-			( props.planProperties || [] ).map( ( properties ) => properties.planName as PlanSlug )
-		);
+		const isLargeCurrency = useIsLargeCurrency( {
+			planSlugs: ( props.planProperties || [] ).map(
+				( properties ) => properties.planName as PlanSlug
+			),
+			siteId: props.siteId,
+		} );
 		return <Component { ...props } isLargeCurrency={ isLargeCurrency } />;
 	};
 };
```

Comment:
Interesting! Unrelated to this PR but I would like to dig into your thoughts behind 

>  I don't think the HOC should live in the same file as the hook.

If this HOC wasn't temporary would you think of it the same way?

Reply:
Hey @jdc91. Asking the cool questions 😄  

> If this HOC wasn't temporary would you think of it the same way?

I believe so. So, I would rather not export a HOC (say, `withFoo`) from a directory of hooks `../hooks/use-foo.ts`, I guess in the same way that I'd not export a hook from a directory of selectors `../state/selectors/getFoo.ts`. 🤔 However, I can see that being part of a general design if done consistently (but is it ever?). I would not go out of my way to object, but I'd maybe raise it/suggest it. 

There are also times when I think it's ok for things to live next to each other, especially if interdepended or part of some closed/isolated scope. An example I recall is this [context file](https://github.com/Automattic/wp-calypso/blob/trunk/packages/tour-kit/src/components/tour-kit-context.tsx) exporting both a provider and a hook to access it. I'd throw the HOC in there too. One could argue if even those could be broken out and exported from an index file (a bit too much IMO).

NITs? What do you think? :-)

Reply:
Right the way I mostly saw it is that the HOC was mainly targeted as a Class Component's way to access the given hook 😁. So in the interest of cohesion, having it closer to the hook as much as possible looked fitting. But I do see your point that it looks a bit unorthodox 😬 

> NITs?
True! But personally love to have these NIT conversations because I learn a lot from them 😀

Reply:
I guess even unorthodox ways can feel ok if done consistently and don't harm discoverability